### ISSUE TITLE
GraalVM 22.2 llegalAccessError: class.... in unnamed module 

### ISSUE THREAD

**Issue by sdelamo** [micronaut-projects/micronaut-core#7738]
Every test project in the GraalVM Test suite 3.6.x-dev which [uses GraalVM branch `release/graal-vm/22.2`](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/commit/86060aabf52133091befc3148bde1a24adc2443e)  fails with: 

```
86Fatal error: java.lang.IllegalAccessError: class io.micronaut.buffer.netty.NettyFeature (in unnamed module @0x684ce74c) cannot access class com.oracle.svm.core.jdk.SystemPropertiesSupport (in module org.graalvm.nativeimage.builder) because module org.graalvm.nativeimage.builder does not export com.oracle.svm.core.jdk to unnamed module @0x684ce74c
[87](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2769223309#L87)	at io.micronaut.buffer.netty.NettyFeature.beforeAnalysis(NettyFeature.java:66)
[88](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2769223309#L88)	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$9(NativeImageGenerator.java:722)
[89](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2769223309#L89)	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:78)
[90](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2769223309#L90)	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:722)
[91](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2769223309#L91)	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:564)
[92](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2769223309#L92)	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:521)
[93](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2769223309#L93)	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:407)
[94](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2769223309#L94)	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:585)
[95](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2769223309#L95)	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
```

see: https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2769223309

Test applications use [Micronaut Gradle Plugin 3.5.1](https://github.com/micronaut-graal-tests/micronaut-basic-app/commit/98f74c7f6918314238706e4fc9382d1069b59461)

**Comment by alvarosanchez** [comment]
This was supposedly fixed in https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/507 which was included in 3.5.1 /cc @melix

**Comment by kristofdho** [comment]
@sdelamo This should be reopened. `SystemPropertiesSupport`, the offending class here, is located in the `svm` dependency. This will no longer be published as of GraalVM 22.3, which only leaves the `graal-sdk` dependency, so this class will be unavailable.
The used fix here was to switch back to running on the classpath, which is a deprecated runtime mode and will also be unavailable in a future version.

https://github.com/oracle/graal/discussions/4146
https://github.com/oracle/graal/discussions/4792
https://github.com/oracle/graal/pull/4783

**Comment by kristofdho** [comment]
@sdelamo GraalVM 23.0.0 will ~possibly~ have a public API to replace the direct access to `SystemPropertiesSupport` https://github.com/oracle/graal/issues/4862#issuecomment-1332350041

update: https://github.com/oracle/graal/issues/5557
